### ISSUE TITLE
SelectedEntry use criterion instead of method/uri_pattern

### ISSUE THREAD

**Issue by DronMDF** [DronMDF/zond#72]
Like this.

```c++
if (criteria.satisfy(request)) {
        return entry->process()
)
```

This gives me opportunity to use different criteria: regex/equal/always

**Comment by 0pdd** [comment]
@DronMDF the puzzle [#74](https://github.com/DronMDF/zond/issues/74) is still not solved.

**Comment by 0pdd** [comment]
@DronMDF 4 puzzles [#79](https://github.com/DronMDF/zond/issues/79), [#80](https://github.com/DronMDF/zond/issues/80), [#81](https://github.com/DronMDF/zond/issues/81), [#82](https://github.com/DronMDF/zond/issues/82) are still not solved; solved: [#74](https://github.com/DronMDF/zond/issues/74).

**Comment by 0pdd** [comment]
@DronMDF 3 puzzles [#79](https://github.com/DronMDF/zond/issues/79), [#80](https://github.com/DronMDF/zond/issues/80), [#82](https://github.com/DronMDF/zond/issues/82) are still not solved; solved: [#74](https://github.com/DronMDF/zond/issues/74), [#81](https://github.com/DronMDF/zond/issues/81).

**Comment by 0pdd** [comment]
@DronMDF 4 puzzles [#91](https://github.com/DronMDF/zond/issues/91), [#92](https://github.com/DronMDF/zond/issues/92), [#93](https://github.com/DronMDF/zond/issues/93), [#94](https://github.com/DronMDF/zond/issues/94) are still not solved; solved: [#74](https://github.com/DronMDF/zond/issues/74), [#79](https://github.com/DronMDF/zond/issues/79), [#80](https://github.com/DronMDF/zond/issues/80), [#81](https://github.com/DronMDF/zond/issues/81), [#82](https://github.com/DronMDF/zond/issues/82).

**Comment by 0pdd** [comment]
@DronMDF 4 puzzles [#91](https://github.com/DronMDF/zond/issues/91), [#93](https://github.com/DronMDF/zond/issues/93), [#94](https://github.com/DronMDF/zond/issues/94), [#97](https://github.com/DronMDF/zond/issues/97) are still not solved; solved: [#74](https://github.com/DronMDF/zond/issues/74), [#79](https://github.com/DronMDF/zond/issues/79), [#80](https://github.com/DronMDF/zond/issues/80), [#81](https://github.com/DronMDF/zond/issues/81), [#82](https://github.com/DronMDF/zond/issues/82), [#92](https://github.com/DronMDF/zond/issues/92).

**Comment by 0pdd** [comment]
@DronMDF 4 puzzles [#182](https://github.com/DronMDF/zond/issues/182), [#93](https://github.com/DronMDF/zond/issues/93), [#94](https://github.com/DronMDF/zond/issues/94), [#97](https://github.com/DronMDF/zond/issues/97) are still not solved; solved: [#74](https://github.com/DronMDF/zond/issues/74), [#79](https://github.com/DronMDF/zond/issues/79), [#80](https://github.com/DronMDF/zond/issues/80), [#81](https://github.com/DronMDF/zond/issues/81), [#82](https://github.com/DronMDF/zond/issues/82), [#91](https://github.com/DronMDF/zond/issues/91), [#92](https://github.com/DronMDF/zond/issues/92).

**Comment by 0pdd** [comment]
@DronMDF 4 puzzles [#184](https://github.com/DronMDF/zond/issues/184), [#93](https://github.com/DronMDF/zond/issues/93), [#94](https://github.com/DronMDF/zond/issues/94), [#97](https://github.com/DronMDF/zond/issues/97) are still not solved; solved: [#182](https://github.com/DronMDF/zond/issues/182), [#74](https://github.com/DronMDF/zond/issues/74), [#79](https://github.com/DronMDF/zond/issues/79), [#80](https://github.com/DronMDF/zond/issues/80), [#81](https://github.com/DronMDF/zond/issues/81), [#82](https://github.com/DronMDF/zond/issues/82), [#91](https://github.com/DronMDF/zond/issues/91), [#92](https://github.com/DronMDF/zond/issues/92).
